app/models/manifest_uploader.rb

```diff
@@ -45,9 +50,53 @@ def missing_locations
     @missing_locations ||= location_barcodes.reject { |barcode| locations.key?(barcode.strip) }
   end
 
+  def ordered_location_barcodes
+    @ordered_location_barcodes ||= location_barcodes.select { |barcode| locations[barcode.strip].type == "OrderedLocation" unless locations[barcode.strip].nil? }
+  end
+
+  def ordered_location_rows
+    @ordered_location_rows ||= data.collect.with_index { |item, index| [(index + 2).to_s] + item if ordered_location_barcodes.include?(item.first.strip) }.compact
+  end
```

Comment:
Took me a while to understand this so maybe improvements:
1.  multi-line `do` `end` instead of `{}`
2. define `row_num = index + 2`,
3. define  `location_barcode = item.first.strip` to make it more readable?

Reply:
I've done this, not sure if it looks a bit messy though - could you take another look?

Reply:
Hey Alex, I think it's fine. I get what you mean that it looks a little messy, so I had a little play around to see if I could improve it. 

Do you think it's any better like this, with another method refactored out? Up to you!
```
  def ordered_location_rows
    @ordered_location_rows ||= find_ordered_location_rows
  end

  def find_ordered_location_rows
    rows = []
    data.each_with_index do |row, index|
      location_barcode = row.first.strip
      if ordered_location_barcodes.include?(location_barcode)
        indexed_row = [(index + 2).to_s] + row
        rows.push(indexed_row)
      end
    end
    rows
  end
```